packages/tpetra/core/src/Tpetra_Details_Transfer_decl.hpp

```diff
@@ -75,7 +75,7 @@ class Transfer : public Teuchos::Describable {
 private:
   using execution_space = typename NT::device_type::execution_space;
   using memory_space =
-    ::Tpetra::Details::DefaultTypes::comm_buffer_memory_space<execution_space>;
+    ::Tpetra::Details::DefaultTypes::comm_buffer_memory_space<typename NT::device_type>;
```

Comment:
Here adding `using device_type = typename NT::device_type` would help promote its use instead of `execution_space`.

Reply:
@lucbv if you look at the line right below, we already have a `using device_type`, in this case, the `typename NT::device_type` is just a temporary to setup the real device type we want to be used.

Reply:
OK but I am a bit curious, what is the different between `typename NT::device_type` and `Kokkos::Device<execution_space, memory_space>`?
I simply don't know so in my mind I thought the type declarations would actually look like this:
```
using device_type = typename NT::device_type;
using execution_space = typename device_type::execution_space;
using memory_space = ::Tpetra::Details::DefaultTypes::comm_buffer_memory_space<device_type>;
```
But I am guessing that Tpetra is doing something special in `::Tpetra::Details::DefaultTypes::comm_buffer_memory_space` that changes the `memory_space` used in `NT::device_type`?

Reply:
@lucbv see `core/src/Tpetra_Details_DefaultTypes.hpp`, lines 105-117, and #1088.  I wasn't around for this history, but apparently there are performance issues with using CudaUVMSpace to create buffers for communication, so when we are using a Cuda node, this code forces the local `device_type` to use CudaSpace regardless of the default memory space associated with the Cuda execution space.